src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/TimeSpanExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class TimeSpanExtensions
+    {
+        private static readonly TimeSpan MinBackoff = TimeSpan.FromSeconds(1);
+        private static readonly TimeSpan MaxBackoff = TimeSpan.FromMinutes(10);
+
+        /// <summary>
+        /// This method calculates a random exponential backoff which lies between <see cref="MinBackoff"/> and <see cref="MaxBackoff"/>.
+        /// </summary>
+        /// <param name="interval">The maximum backoff to be used if <paramref name="interval"/> is less than <see cref="MaxBackoff"/>.</param>
+        /// <param name="attempts">The number of attempts made to backoff.</param>
+        /// <returns>The calculated exponential backoff time.</returns>
+        public static TimeSpan CalculateBackoffTime(this TimeSpan interval, int attempts)
+        {
+            if (attempts < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(attempts), attempts, "The number of attempts should not be less than 1.");
+            }
+
+            if (interval < MinBackoff)
+            {
+                throw new ArgumentOutOfRangeException(nameof(interval), interval, $"The interval cannot be less than {nameof(MinBackoff)}.");
```

Comment:
The `internal` is the cache expiration time a user sets. Now that you throw if it's smaller than `MinBackOff`. This is exactly what we are already doing in `AzureAppConfigurationRefreshOptions.SetCacheExpiration`. The `MinBackoff` shouldn't be different than `MinimumCacheExpirationInterval`.

Reply:
Now I think about it again - this whole exponential backoff is to prevent stale cache in case a user sets a very long expiration and the last attempt fails. This is not a problem if the user sets a reasonable short expiration time. Therefore, I think our `MinBackoff` should be something higher. It can be our default cache expiration time, i.e., 30s or even higher like 1 min.

Reply:
If the refresh interval set by users is short (say 5 seconds), are we saying we still won't refresh for another 1 minute?
The idea was to have a backoff time between `[MinimumCacheExpirationInterval, Math.Min(cacheExpirationInterval, DefaultMaxBackoff)]`.

Reply:
Correct. The underlying SDK retries in a much shorter time automatically already. 

You introduce a second layer of retry to solve the long expiration time issue. In your example, your new backoff retry won't kick in at all. If something keeps failing, we will retry every 5 seconds as the user configured.

Reply:
Here is what I think the backoff time should be. Let's say a user sets cacheExpiration

- if (cacheExpiration < minBackoff), we honor cacheExpiration
- if (minBackoff < cacheExpiration < maxBackoff), we pick a value between [minBackoff, cacheExpiration]
- if (cacheExpiration > maxBackoff), we use maxBackoff

Reply:
Sounds good, let's use 30 seconds min backoff time then. Its consistent with our DefaultCacheExpirationInterval.
Also, one correction to the last point:

- if (cacheExpiration > maxBackoff), we pick a value between [minBackoff, maxBackoff]

Reply:
Updated.